Makefile

```diff
@@ -148,12 +150,22 @@ uninstall:
 	rm -f $(BINDIR)/ctr
 
 $(BUILD_DIR)/$(TARBALL): static-binaries vendor.conf
-	@BUILD_DIR=$(BUILD_DIR) TARBALL=$(TARBALL) VERSION=$(VERSION) ./hack/release.sh
+	@BUILD_DIR=$(BUILD_DIR) TARBALL=$(join $(TARBALL),-$(ARCH).tar.gz) VERSION=$(VERSION) ./hack/release.sh
 
-release: $(BUILD_DIR)/$(TARBALL)
+release: all-release
 
-push: $(BUILD_DIR)/$(TARBALL)
-	@BUILD_DIR=$(BUILD_DIR) TARBALL=$(TARBALL) VERSION=$(VERSION) ./hack/push.sh
+sub-release-%:
+	$(MAKE) --no-print-directory ARCH=$* $(BUILD_DIR)/$(TARBALL)
+
+all-release: $(addprefix sub-release-,$(ALL_ARCH))
+
+push: all-push
```

Comment:
we don't need all-push target here, we can use $(addprefix sub-release-push-,$(ALL_ARCH)) here.